doc/source/administrator/optimization.md

```diff
@@ -383,3 +383,133 @@ Choosing the right _limit to guarantee ratio_ ratio is an art, not a science.
+| proxy.secretSync.resources              | autohttps                     | -                                | The sidecar container is a watchdog, watching a file for changes and updates a k8s Secret with those changes. Will require minimal resources.                                                                                                                                                                                      |
+| scheduling.userScheduler.resources      | user-scheduler                | 50m, 256Mi                       | The container runs a `kube-scheduler` binary with custom configuration to schedule the user pods. Will require a small amount of resources.                                                                                                                                                                                        |
+| scheduling.userPlaceholder.resources    | user-placeholder              | -                                | This is an explicit override of the default behavior to reuse the values in `singleuser.cpu.guarantee\|limit` and `singleuser.memory.guarantee\|limit`. It can be useful to increase this to a multiple of the typical real users' requests if you want to have may user-placeholder pods to reduce the pod scheduling complexity. |
+| prePuller.resources                     | hook\|continuous-image-puller | 0, 0                             | This pod's containers are all running `echo` or `pause` commands as a trick to pull the images. Will require minimal resources.                                                                                                                                                                                                    |
+| prePuller.hook.resources                | hook-image-awaiter            | 0, 0                             | The container just polls the k8s api-server. Will require minimal resources.                                                                                                                                                                                                                                                       |
+| singleuser.cpu\|memory.guarantee\|limit | jupyter-username              | 0, 1G                            | The configuration syntax is different because it is native to the Spawner base class rather than Kubernetes. It is commonly useful to guarantee a certain amount of memory rather than CPU to help users share CPU with each other.                                                                                                |
+
+### Example resource requests for high reliability
+
+Here are some examples of resource requests reasonable for a larger deployment.
+If you collect metrics on the resource usage of pods in your k8s cluster
+(Prometheus) and have dashboards showing you its usage (Grafana), you could
+tweak these over time.
```

Comment:
What counts as a "larger deployment"? If it's mybinder.org you could say that, and also link to the relevant mybinder.org grafana dashboards?

Reply:
I'm not sure, I guess I meant "if you are so large that you don't care about allocating 1 CPU to the hub and proxy pod to ensure stability over minimizing cost".

Reply:
I rewrote it as the following in 041d37f

> Here are some examples of resource requests that could be reasonable for adeployment where service reliability and performance matters more thanminimizing the required capacity on nodes.